Unable to delete images from ECR

**Expected behavior and actual behavior:**
Harbor 1.9.3 is setup to replicate images to ECR (and to delete images if these are deleted from Harbor). Replication works fine, but when I delete a tag, the tag is deleted from Harbor but deletion from ECR fails:

```
2019-12-14T10:53:33Z [INFO] [/replication/transfer/image/transfer.go:95]: client for source registry [type: harbor, URL: http://harbor-harbor-core, insecure: true] created
2019-12-14T10:53:33Z [INFO] [/replication/transfer/image/transfer.go:105]: client for destination registry [type: aws-ecr, URL: https://api.ecr.us-west-2.amazonaws.com, insecure: false] created
2019-12-14T10:53:33Z [ERROR] [/replication/transfer/image/transfer.go:367]: failed to delete the manifest of image XXX/releases/pause:3.1 on the destination registry: http error: code 404, message 404 page not found
```

Verified that the image existed in ECR:

```
$ aws ecr list-images  --repository-name 'XXX/releases/pause' --region us-west-2
{
     "imageIds": [
        {
            "imageTag": "3.1",
            "imageDigest": "sha256:fcaff905397ba63fd376d0c3019f1f1cb6e7506131389edbcb3d22719f1ae54d"
        }
    ]
}
```

Also verified that the image can be deleted using the same AWS credentials as Harbor:
```
$ aws ecr batch-delete-image --repository-name XXX/releases/pause --image-ids imageTag=3.1
{
    "failures": [],
    "imageIds": [
        {
            "imageTag": "3.1",
            "imageDigest": "sha256:fcaff905397ba63fd376d0c3019f1f1cb6e7506131389edbcb3d22719f1ae54d"
        }
    ]
}

$ aws ecr list-images  --repository-name 'XXX/releases/pause' --region us-west-2 
{
    "imageIds": []
}
```


**Steps to reproduce the problem:**

- Setup replication to ECR for one repository
- Push a new image/tag to that repository => tag is successfully replicated to ECR 
- Delete tag from Harbor => 404 Not found, image is not deleted from ECR

**Versions:**
Please specify the versions of following systems.

- harbor version: 1.9.3

--- a/src/replication/adapter/awsecr/adapter.go
+++ b/src/replication/adapter/awsecr/adapter.go
@@ -181,3 +181,38 @@ func (a *adapter) createRepository(repository string) error {
 	}
 	return nil
 }
+
+// DeleteManifest ...
+func (a *adapter) DeleteManifest(repository, reference string) error {
+	// AWS doesn't implement standard OCI delete manifest API, so use it's sdk.
+	if a.registry.Credential == nil ||
+		len(a.registry.Credential.AccessKey) == 0 || len(a.registry.Credential.AccessSecret) == 0 {
+		return errors.New("no credential ")
+	}
+	cred := credentials.NewStaticCredentials(
+		a.registry.Credential.AccessKey,
+		a.registry.Credential.AccessSecret,
+		"")
+	if a.region == "" {
+		return errors.New("no region parsed")
+	}
+	config := &aws.Config{
+		Credentials: cred,
+		Region:      &a.region,
+		HTTPClient: &http.Client{
+			Transport: registry.GetHTTPTransport(a.registry.Insecure),
+		},
+	}
+	if a.forceEndpoint != nil {
+		config.Endpoint = a.forceEndpoint
+	}
+	sess := session.Must(session.NewSession(config))
+
+	svc := awsecrapi.New(sess)
+
+	_, err := svc.BatchDeleteImage(&awsecrapi.BatchDeleteImageInput{
+		RepositoryName: &repository,
+		ImageIds:       []*awsecrapi.ImageIdentifier{{ImageTag: &reference}},
+	})
+	return err
+}

